### ISSUE TITLE
State is kept in library, subsequent calls fail previous ones

### ISSUE THREAD

**Issue by tswaters** [flexdinesh/typy#4]
From the reddit thread announcing the module,

Input:

```js
var t = require('typy')
var stringType = t('hello')
console.log('stringType.isString', stringType.isString)
var numberType = t(123)
console.log('numberType.isNumber', numberType.isNumber)
// Original types are now broken!
console.log('stringType.isString', stringType.isString)
console.log('stringType.isNumber', stringType.isNumber)
```

Output:

```js
stringType.isString true
numberType.isNumber true
stringType.isString false
stringType.isNumber true
```

**Comment by flexdinesh** [comment]
Thanks for the catch. This should be handled by creating a new instance for each input.

**Comment by flexdinesh** [comment]
@tswaters This issue has been addressed in version 1.2.0. Thanks much for the catch.
